mdformat/plugins.py

```diff
@@ -15,3 +20,33 @@ def _load_codeformatters() -> Dict[str, Callable[[str, str], str]]:
 
 
 CODEFORMATTERS: Mapping[str, Callable[[str, str], str]] = _load_codeformatters()
+
+
+class ParsePluginAbstract:
```

Comment:
Is it intended that plugins inherit from this? Maybe make this a `abc.ABC` and make the methods `@abstractmethod`s.

Reply:
yeh I wasn't sure about that. I don't think its really necessary to inherit from this as it wouldn't add any extra functionality to do so. You should even actually be able to point the entry point to a module, containing the two functions (hence why they are set as staticmethods)
This was primarily for use in typing, unless there is a better to do this?

Reply:
I think https://docs.python.org/3/library/typing.html#typing.Protocol might be the appropriate class here, as that doesn't imply inheritance, it simply tells the type checker what interface should be expected. The annoying thing is that in Python <3.8 you would have to import that from https://pypi.org/project/typing-extensions/.

I would probably just be lazy and typehint as `typing.Any` (and remove this abstract class), but also if you want to do it, I have nothing against a `Protocol` class (in which case you have to add something like `typing-extensions= { version = "*", python = "<3.8" }` to `pyproject.toml`.